server/src/main/java/com/ncc/neon/better/NlpModule.java

```diff
@@ -17,54 +25,81 @@
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
-import java.util.Map;
 
 /*
 Abstract class to represent a remote NLP module REST service.  These services typically expose endpoints to perform
 NLP operations for preprocessing, training, and inference.
  */
 public abstract class NlpModule {
     private String name;
+    private HttpEndpoint statusEndpoint;
     private WebClient client;
-    private HttpEndpoint[] endpoints;
     private DatasetService datasetService;
     private FileShareService fileShareService;
     private BetterFileService betterFileService;
+    private ModuleService moduleService;
+    private Environment env;
 
-    NlpModule(DatasetService datasetService, FileShareService fileShareService, BetterFileService betterFileService) {
+    NlpModule(NlpModuleModel moduleModel, DatasetService datasetService, FileShareService fileShareService, BetterFileService betterFileService, ModuleService moduleService, Environment env) {
         this.datasetService = datasetService;
         this.fileShareService = fileShareService;
         this.betterFileService = betterFileService;
+        this.moduleService = moduleService;
+        this.env = env;
+        name = moduleModel.getName();
+        client = buildNlpWebClient(name);
+        initEndpoints(moduleModel.getEndpoints());
     }
 
+    public String getName() { return this.name; }
+
     public void setName(String name) {
         this.name = name;
     }
 
-    public void setClient(WebClient client) {
-        this.client = client;
+    public Mono<HttpStatus> getRemoteStatus() {
+        return buildRequest(new HashMap<>(), statusEndpoint)
+                .accept(MediaType.APPLICATION_JSON)
+                .retrieve()
+                .bodyToMono(HttpStatus.class)
+                .doOnSuccess(res -> moduleService.getById(name)
+                            .flatMap(module -> {
+                                // Only update status if it's changed.
+                                if (module.getStatus().equals(ModuleStatus.DOWN.toString())) {
+                                    return moduleService.setStatusToActive(name);
```

Comment:
I don't understand this. First, it looks like it could definitely be updated even if it's not changed (i.e., if it's still DOWN), and second, this tells it to set its status to active if it's down? Huh?

Reply:
This is occurring in the `doOnSuccess`, which means the NLP module is alive.  Which means we only want to update the status if it is currently set do down, setting the new current status to active.  Conversely, if we get a success, and the current status is active, we don't want to do anything.

Reply:
Ah, ok.